conan_support.cmake

```diff
@@ -39,6 +39,22 @@ function(detect_compiler COMPILER COMPILER_VERSION)
     if(_COMPILER MATCHES MSVC)
         set(_COMPILER "msvc")
         string(SUBSTRING ${MSVC_VERSION} 0 3 _COMPILER_VERSION)
+        # Configure compiler.runtime and compiler.runtime_type settings for MSVC
+        if (CMAKE_MSVC_RUNTIME_LIBRARY)
```

Comment:
Perhaps for consistency I'd consider a `DEFINED` here, as in line 25? Although it does not make a practical difference in this case.

On the other hand, I believe this conditional does not cover the most common use-case, that is, leaving CMake to pick-up the default "Visual Studio" generator, which is multi-configuration. For such a generator, I'd expect that if `CMAKE_MSVC_RUNTIME_LIBRARY` is defined, it is conditional on the build type, e.g.  `MultiThreaded$<$<CONFIG:Debug>:Debug>`  (as per documentation [here](https://cmake.org/cmake/help/latest/variable/CMAKE_MSVC_RUNTIME_LIBRARY.html)) - and because of the use of `STREQUAL` this is not covered here. 


If a project specified `MultiThreaded$<$<CONFIG:Debug>:Debug>` (the `static` compiler runtime) - it would not be covered by this conditional, we would then only pass `compiler.build_type` as per https://github.com/conan-io/cmake-conan/blob/a261d04f3ce117ec31accb0795b93c7f8a86e452/conan_support.cmake#L207C19-L209, which may fall back to the default `dynamic` runtime, causing a mismatch.

On the other hand, if a project using a multi-config generator like Visual Studio specifies a single fixed value (e.g. `MultiThreadedDebugDLL`), that would be used also for the release configuration, and I'd imagine there may be binary incompatibilities.

Perhaps a way to make this more robust - we need to consider two axes:
* whether `CMAKE_MSVC_RUNTIME_LIBRARY` is defined, and if it is, whether it contains a generator expression (`$<`) 
* whether we are in a single or multi configuration generator (testing the presence of `CMAKE_CONFIGURATION_TYPES`)


If we are in a single configuration generator:
      * and `CMAKE_MSVC_RUNTIME_LIBRARY` is defined and matches one of the four values in this list (no generator expression), then it makes sense to define `compiler.runtime` and `compiler.runtime_type` with this exact mapping)
      * if `CMAKE_MSVC_RUNTIME_LIBRARY` is defined and contains a generator expression (we could test for the specific $<CONFIG: string) - then we need to test for the presence of `DLL` in the string to define `compiler.runtime`, and let Conan derive the `runtime_type` based on the `build_type` setting (as per https://github.com/conan-io/cmake-conan/blob/a261d04f3ce117ec31accb0795b93c7f8a86e452/conan_support.cmake#L88-L92)
      
If we are in a multi configuration generator:
    * If `CMAKE_MSVC_RUNTIME_LIBRARY` is one of these four values with no generator expression, this is likely an error - it's almost certain one of the configurations will not build at all due to incompatibilities
    * If `CMAKE_MSVC_RUNTIME_LIBRARY` is defined and contains a generator expression on the build type, same as before: evaluate the presence of `DLL` to determine the `runtime_type`, and let the `build_type` derive the rest (as per https://github.com/conan-io/cmake-conan/blob/a261d04f3ce117ec31accb0795b93c7f8a86e452/conan_support.cmake#L207-L209)

Reply:
Thank you @jcar87 for detailed response! I totally agree, going to try your approach tomorrow.